app/src/main/java/org/fossasia/openevent/activities/MainActivity.java

```diff
@@ -254,6 +281,36 @@ private void doMenuAction(int menuItemId){
         mDrawerLayout.closeDrawers();
     }
 
+    public class NetworkDialogHandler extends BroadcastReceiver{
+        private boolean isConnected = false;
+        @Override
+        public void onReceive(Context context, Intent intent) {
+            Log.d("NetNotif", "Received notif about net status");
+            isNetworkAvailable(context);
+        }
+
+        private boolean isNetworkAvailable(Context context){
+            ConnectivityManager connectivityManager = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
+            NetworkInfo networkInfo = connectivityManager.getActiveNetworkInfo();
+            if(networkInfo!=null){
+                if(networkInfo.getState().equals(NetworkInfo.State.CONNECTED)){
+                    if(!isConnected){
+                        isConnected=true;
+                        DataDownload download = new DataDownload();
+                        download.downloadVersions();
+                        Log.d("NetNotif", "Connected to Internet");
+
+                    }
+                    return true;
+                }
+            }
+            Log.d("NetNotif", "Not connected to Internet");
+            showNetworkDialog(MainActivity.this);
```

Comment:
you cannot call view or activity related code from a BroadcastReceiver. This is wrong usage,